go/runtime/capture.go

```diff
@@ -190,7 +192,12 @@ func pollLoop(
 			}
 			return err
 
-		case <-pollCh:
+		case op = <-pollCh:
+		}
+
+		// Wait for the prior commit's OpFuture to resolve succesfully.
+		if err := op.Err(); err != nil {
```

Comment:
I assume it's intentional that you're not doing a second select between termCtx and op.Done() here?

Reply:
yea, the shard context is rolled into op.Done(), and if the term context is cancelled we'll get it on a future round.